nmdc_runtime/site/ops.py

```diff
@@ -872,3 +905,41 @@ def get_csv_rows_from_url(url: Optional[str]) -> List[Dict]:
     # Collect all the rows into a list of dicts while stripping out (valfilter) cells where the
     # value is an empty string (identity returns a Falsy value).
     return [valfilter(identity, row) for row in reader]
+
+
+@op
+def get_df_from_url(url: str) -> pd.DataFrame:
+    """Download and return a pandas DataFrame from the URL of a TSV file.
+
+    :param url: raw URL of the TSV file to be downloaded as a DataFrame
+    :return: pandas DataFrame of TSV data
+    """
+    if not url:
+        return pd.DataFrame()
+
+    response = requests.get(url)
+    response.raise_for_status()
+
+    # Using Pandas read_csv to directly read the file-like object
+    df = pd.read_csv(url, delimiter="\t")
+
+    return df
+
+
+@op
+def site_code_mapping() -> dict:
```

Comment:
Separate `@op` that returns a dictionary of site codes (as key) and their constructed `geo_loc_name` (as value).

In response to: https://github.com/microbiomedata/nmdc-runtime/pull/308#discussion_r1416102634